daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala

```diff
@@ -439,6 +444,35 @@ private[validation] object Typing {
         KStar
     }
 
+    private def expandTypeSynonyms(typ0: Type): Type = typ0 match {
+      case TSynApp(syn, args) =>
+        expandSynApp(syn, args)
```

Comment:
In the Haskell typechecker, you recurse on the result of the expansion to expand synonyms that might be in there. Is this not necessary here? If so why?

Reply:
It is necessary here also. Thanks, nice catch. Fixed now.